### PR TITLE
os: add gc job

### PR DESCRIPTION
**Pull request by arithx**
Adds a job to run the ore gc command on all platforms.

Depends on: https://github.com/coreos/mantle/pull/783

### CODE REVIEW

**Comment by arithx** [comment]
Mantle dependency has been merged.

File: os/gc.groovy [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        credentials(credentialType: 'org.jenkinsci.plugins.plaincredentials.impl.FileCredentialsImpl',
+                    defaultValue: '787e229d-0941-4f04-aba5-9e5f22fe1c71',
+                    description: 'Credentials to create DigitalOcean droplets',
+                    name: 'DIGITALOCEAN_CREDS',
+                    required: true),
+        credentials(credentialType: 'org.jenkinsci.plugins.plaincredentials.impl.FileCredentialsImpl',
+                    defaultValue: 'jenkins-coreos-systems-write-5df31bf86df3.json',
+                    description: '''Credentials ID for a JSON file passed as \
+the GOOGLE_APPLICATION_CREDENTIALS value for uploading release files to the \
+Google Storage URL, requires write permission''',
```
Comment (bgilbert) [inline os/gc.groovy]: Does this really need release creds, or only creds sufficient to run GCE instances?

Comment (arithx) [reply os/gc.groovy]: I'm going to update the description to match what's actually required a little bit better. The `gcloud` credentials should just need to be able to auth & potentially create a bucket (if one does not exist at the given Storage URL).

I'm using the same `defaultValue` for the credentials as the `DOWNLOAD_CREDS` from the `os/kola/packet` job.

File: os/gc.groovy [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        string(name: 'PACKET_PROJECT',
+               defaultValue: '9da29e12-d97c-4d6e-b5aa-72174390d57a',
+               description: 'The Packet project ID to run test machines'),
+        string(name: 'UPLOAD_ROOT',
+               defaultValue: 'gs://builds.developer.core-os.net',
+               description: 'URL prefix where development files are uploaded'),
+    ])
+])
+
+node('coreos && amd64 && sudo') {
```
Comment (euank) [inline os/gc.groovy]: I don't think this actually needs coreos or sudo; I think it can just be `node('amd64')`

File: os/gc.groovy [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    stage('Build') {
```
Comment (euank) [inline os/gc.groovy]: s/Build/GC/ ?

**Comment by arithx** [comment]
Updated to address nits

File: os/gc.groovy [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                    required: true),
+        credentials(credentialType: 'org.jenkinsci.plugins.plaincredentials.impl.FileCredentialsImpl',
+                    defaultValue: 'jenkins-coreos-systems-write-5df31bf86df3.json',
+                    description: '''Credentials ID for a JSON file passed as \
+the GOOGLE_APPLICATION_CREDENTIALS value for authing to gcloud & creating a bucket''',
+                    name: 'GS_CREDS',
+                    required: true),
+        credentials(credentialType: 'org.jenkinsci.plugins.plaincredentials.impl.FileCredentialsImpl',
+                    defaultValue: '17942f6a-969b-466d-ac46-cd15925c8953',
+                    description: 'Config required by "kola run --platform=oci"',
```
Comment (euank) [inline os/gc.groovy]: The two `kola run` comments are stale; techncially these can be less permissive credentials (although realistically there's not much benefit splitting out stricter credentials).

This comment also applies to other descriptions, such as packet's and do's.

Comment (arithx) [reply os/gc.groovy]: Pushed an update to reference `ore <platform> gc` and unified the descriptions for `packet` & `do` to match other platforms.

**Pull request merged**